packages/client/hmi-client/src/components/asset/tera-asset.vue

```diff
@@ -166,6 +171,23 @@ function onScroll(event: Event) {
 	chosenItem.value = closestItem;
 }
 
+function onRename(newName: string) {
+	if (!newName) return;
+	switch (pageType) {
+		case ProjectPages.OVERVIEW:
+			if (!activeProject.value) return;
+			updateProject({ ...activeProject.value, name: newName }).then(() => {
+				useProjects().refresh();
+			});
+			break;
+		case AssetType.Dataset:
+		case AssetType.Model:
+		default:
+			emit('rename', newName);
```

Comment:
model and dataset require saving an entire object since we can modify many parts within those asset pages, so im emitting the rename here.

Reply:
That make sense, in that case I would prefer all pageType to emit a `rename` event, and let each page handle how this is done. So in this case, the project page should handle the renaming, not tera-asset. The goal being to simplify tera-asset and remove as many condition in it.

Reply:
done, I've moved that snippet to the project overview